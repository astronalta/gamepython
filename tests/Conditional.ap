Conditional < Object {

    conditional1() Int {
        if true {
            return 1
        } else {
            return 0
        }
    }

    conditional2() Int {
        if true {
            return 1
        } else if false {
            return 2
        } else {
            return 3
        }
    }

    conditional3() Int {
        if true {
            return 1
        } else {
            if true {
                return 2
            } else if false {
                return 4
            } else {
                return 6
            }
        }
    }
}
