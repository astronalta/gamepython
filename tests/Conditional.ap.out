Module
  name: Conditional
  feature0: Class
    name: Conditional
    feature0: Function
      name: conditional1
      type: Int
      formal0: Formal
        name: self
        type: Self
      block: Block
        child0: Conditional
          guard: BooleanLiteral(true)
          true: Block
            child0: Return
              expression: IntegerLiteral(1)
          false: Block
            child0: Return
              expression: IntegerLiteral(0)
    feature1: Function
      name: conditional2
      type: Int
      formal0: Formal
        name: self
        type: Self
      block: Block
        child0: Conditional
          guard: BooleanLiteral(true)
          true: Block
            child0: Return
              expression: IntegerLiteral(1)
          false: Conditional
            guard: BooleanLiteral(false)
            true: Block
              child0: Return
                expression: IntegerLiteral(2)
            false: Block
              child0: Return
                expression: IntegerLiteral(3)
    feature2: Function
      name: conditional3
      type: Int
      formal0: Formal
        name: self
        type: Self
      block: Block
        child0: Conditional
          guard: BooleanLiteral(true)
          true: Block
            child0: Return
              expression: IntegerLiteral(1)
          false: Block
            child0: Conditional
              guard: BooleanLiteral(true)
              true: Block
                child0: Return
                  expression: IntegerLiteral(2)
              false: Conditional
                guard: BooleanLiteral(false)
                true: Block
                  child0: Return
                    expression: IntegerLiteral(4)
                false: Block
                  child0: Return
                    expression: IntegerLiteral(6)
